zebrad/src/components/tracing/endpoint.rs

```diff
@@ -3,20 +3,27 @@
 use std::net::SocketAddr;
 
 use abscissa_core::{Component, FrameworkError};
-use hyper::service::{make_service_fn, service_fn};
-use hyper::{Body, Request, Response, Server};
 
-use crate::{components::tokio::TokioComponent, config::ZebradConfig, prelude::*};
+use crate::config::ZebradConfig;
 
-use super::Tracing;
+#[cfg(feature = "filter-reload")]
+use hyper::{Body, Request, Response};
+
+#[cfg(feature = "filter-reload")]
+use crate::{components::tokio::TokioComponent, prelude::*};
 
 /// Abscissa component which runs a tracing filter endpoint.
 #[derive(Debug, Component)]
-#[component(inject = "init_tokio(zebrad::components::tokio::TokioComponent)")]
+#[cfg_attr(
+    feature = "filter-reload",
+    component(inject = "init_tokio(zebrad::components::tokio::TokioComponent)")
+)]
 pub struct TracingEndpoint {
+    #[allow(dead_code)]
     addr: Option<SocketAddr>,
```

Comment:
Is this now potentially `dead_code` because these lines https://github.com/ZcashFoundation/zebra/blob/optional-deps/zebrad/src/components/tracing/endpoint.rs#L59-L60 are now under
```
#[cfg(feature = "filter-reload")]
```
?

Reply:
`addr` is always set in `TracingEndpoint::new()`:
https://github.com/ZcashFoundation/zebra/pull/4539/files#diff-a49354aba5c56e2d44d0c10614f94a93cce82a72b3c724d662e8ead279ebc049R48

But when the `filter-reload` feature is off, it's not actually read anywhere.